cdap-ui/app/cdap/components/PreviewData/index.tsx

```diff
@@ -97,26 +105,42 @@ const PreviewDataViewBase: React.FC<IPreviewDataViewProps> = ({
   const tableData: ITableData = getTableData();
 
   const loadingMsg = (cls) => (
-    <div className={classes.headingContainer}>
-      <Heading type={HeadingTypes.h3} label={'Fetching preview data'} className={cls.messageText} />
+    <div className={cls.headingContainer}>
+      <Heading
+        type={HeadingTypes.h3}
+        label={T.translate(`${I18N_PREFIX}.loading`)}
+        className={cls.messageText}
+      />
       <LoadingSVGCentered />
     </div>
   );
 
   const noPreviewDataMsg = (cls) => (
-    <div className={classes.headingContainer}>
+    <div className={cls.headingContainer}>
+      <Heading
+        type={HeadingTypes.h3}
+        label={T.translate(`${I18N_PREFIX}.runPreview`)}
+        className={cls.messageText}
+      />
+    </div>
+  );
+
+  const errorMsg = (cls) => (
+    <div className={classnames('text-danger', cls.headingContainer)}>
       <Heading
         type={HeadingTypes.h3}
-        label={'Run preview to generate preview data.'}
+        label={T.translate(`${I18N_PREFIX}.errorHeader`)}
         className={cls.messageText}
       />
+      <span>{error}</span>
```

Comment:
Since we are not sure where the error message will reside (error.response or errror.response.message) can we check if its an object and avoid rendering object in the DOM? It will break the entire app.